src/Type/Php/ReplaceFunctionsDynamicReturnTypeExtension.php

```diff
@@ -92,68 +103,89 @@ private function getPreliminarilyResolvedTypeFromFunctionCall(
 				if ($replaceArgumentType->isArray()->yes()) {
 					$replaceArgumentType = $replaceArgumentType->getIterableValueType();
 				}
+			}
+		}
 
-				$accessories = [];
-				if ($subjectArgumentType->isNonFalsyString()->yes() && $replaceArgumentType->isNonFalsyString()->yes()) {
-					$accessories[] = new AccessoryNonFalsyStringType();
-				} elseif ($subjectArgumentType->isNonEmptyString()->yes() && $replaceArgumentType->isNonEmptyString()->yes()) {
-					$accessories[] = new AccessoryNonEmptyStringType();
-				}
+		$result = [];
 
-				if ($subjectArgumentType->isLowercaseString()->yes() && $replaceArgumentType->isLowercaseString()->yes()) {
-					$accessories[] = new AccessoryLowercaseStringType();
-				}
+		$stringArgumentType = TypeCombinator::intersect(new StringType(), $subjectArgumentType);
+		if (!$stringArgumentType instanceof NeverType) {
+			$result[] = $this->getReplaceType($stringArgumentType, $replaceArgumentType);
 
-				if ($subjectArgumentType->isUppercaseString()->yes() && $replaceArgumentType->isUppercaseString()->yes()) {
-					$accessories[] = new AccessoryUppercaseStringType();
-				}
+			$subjectArgumentType = $subjectArgumentType->tryRemove($stringArgumentType);
+		}
 
-				if (count($accessories) > 0) {
-					$accessories[] = new StringType();
-					return new IntersectionType($accessories);
+		if ($subjectArgumentType->isArray()->yes()) {
```

Comment:
When we get `string|array<string>`

I didn't find a better way:
- First, to work with the `string` part first (in order to know if it's non-empty, lowercase, etc)
- Secondly to work only with the `array` part (in order to know the shape, if it's a list, non empty, the values, etc)

To access the string part, the best I found was 
```
$stringArgumentType = TypeCombinator::intersect(new StringType(), $subjectArgumentType);
```

To access the array part, the best I found was 
```
$subjectArgumentType = $subjectArgumentType->tryRemove($stringArgumentType);
```
because
- $subjectArgumentType->getArrays() does not return array on union
- TypeUtiles::toBenevolent($subjectArgumentType)->getArrays() is returning arrays but is loosing the non-empty and the list accessory.